src/cli/init.ts

```diff
@@ -0,0 +1,189 @@
+import { execSync } from "child_process";
+import { existsSync, mkdirSync, readdirSync, rmSync, statSync } from "fs";
+import path from "path";
+import process from "process";
+
+import { Command } from "@commander-js/extra-typings";
+import { confirm, input, select } from "@inquirer/prompts";
+
+import { logger } from "cli/logging";
+import { scaffoldDirectory } from "cli/utils";
+
+export const initCommand = new Command()
+  .name("init")
+  .description("Initialize a new Sindri project.")
+  .argument(
+    "[directory]",
+    "The directory where the new project should be initialized.",
+    ".",
+  )
+  .action(async (directory) => {
+    // Prepare the directory paths.
+    const directoryPath = path.resolve(directory);
+    const directoryName = path.basename(directoryPath);
+
+    // Ensure that the directory exists.
+    if (!existsSync(directoryPath)) {
+      mkdirSync(directoryPath, { recursive: true });
+    } else if (!statSync(directoryPath).isDirectory()) {
+      logger.warn(
+        `File "${directoryPath}" exists and is not a directory, aborting.`,
+      );
+      return process.exit(1);
+    }
+
+    // Check that the directory is empty.
+    const existingFiles = readdirSync(directoryPath);
+    if (existingFiles.length > 0) {
+      const proceed = await confirm({
+        message:
+          `The "${directoryPath}" directory already exists and contains files. Continuing will ` +
+          "overwrite your files. Are you *SURE* you would like to proceed?",
+        default: false,
+      });
+      if (!proceed) {
+        logger.info("Aborting.");
+        return process.exit(1);
+      } else {
+        rmSync(directoryPath, { recursive: true, force: true });
```

Comment:
I added this because without deleting the old directory, I ran into the following issue:
1) `sindri init gnark-test` with package name `gnarkTest`
2) Run `sindri init gnark-test` again, but set package name to `katie`
3) Sindri compilation complains that we've got two packages in the upload: both `gnarkTest.go` and `katie.go`

Reply:
I feel like this is unexpected and not how most tools I'm familiar with typically operate. It's not uncommon to have some progress in a directory, like perhaps you've already initialized the git repo? Running with that example, the later code for git initialization specifically checks if there's already an existing repo. It's unfortunate that the compilation fails there, but it's far worse to delete a users files unexpectedly. At the very least, the error message would need to be much more explicit about it.